homeassistant/components/solarlog/sensor.py

```diff
@@ -0,0 +1,150 @@
+"""Platform for solarlog sensors."""
+import logging
+
+from requests.exceptions import HTTPError, Timeout
+from sunwatcher.solarlog.solarlog import SolarLog
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.config_entries import SOURCE_IMPORT
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+from .const import DOMAIN, DEFAULT_HOST, DEFAULT_NAME, SCAN_INTERVAL, SENSOR_TYPES
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Import YAML configuration when available."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=dict(config)
+        )
+    )
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Add solarlog entry."""
+    host = entry.data[CONF_HOST]
+    platform_name = entry.title
+    try:
+        api = SolarLog(f"http://{host}")
+        _LOGGER.debug("Connected to Solar-Log device, setting up entries")
+    except (OSError, HTTPError, Timeout):
+        _LOGGER.error(
+            "Could not connect to Solar-Log device at %s, check host ip address", host
+        )
+        return
+
+    # Create solarlog data service which will retrieve and update the data.
+    data = SolarlogData(hass, api, host)
```

Comment:
If this is doing I/O, run it in executor.

Reply:
Fixed